### DISCUSSION TITLE
getServerSideProps BUG   Very serious

### DISCUSSION THREAD

**Discussion by parlay96** [vercel/next.js#34262] [Help]
### What version of Next.js are you using?

12.0

### What version of Node.js are you using?

14

### What browser are you using?

chrome

### What operating system are you using?

Windows

### How are you deploying your application?

next start

### Describe the Bug

The a page uses getserversideprops, router push（'B'）
Then, if there is a scroll bar on page B,
Use the browser return button to return to page a. Page B will flash, the scroll bar will scroll down, and then go to page a.

The effect is not good

### Expected Behavior

The a page uses getserversideprops, router push（'B'）
Then, if there is a scroll bar on page B,
Use the browser return button to return to page a. Page B will flash, the scroll bar will scroll down, and then go to page a.

The effect is not good

### To Reproduce

The a page uses getserversideprops, router push（'B'）
Then, if there is a scroll bar on page B,
Use the browser return button to return to page a. Page B will flash, the scroll bar will scroll down, and then go to page a.

The effect is not good

**Comment by parlay96** [comment]
Request answer，thank you

**Reply by SamRoehrich** [reply]
The bug and expected behavior are the same thing.

**Comment by parlay96** [comment]
@SamRoehrich what do you mean

**Reply by SamRoehrich** [reply]
```
Describe the Bug
The a page uses getserversideprops, router push（'B'）
Then, if there is a scroll bar on page B,
Use the browser return button to return to page a. Page B will flash, the scroll bar will scroll down, and then go to page a.

The effect is not good

Expected Behavior
The a page uses getserversideprops, router push（'B'）
Then, if there is a scroll bar on page B,
Use the browser return button to return to page a. Page B will flash, the scroll bar will scroll down, and then go to page a.

The effect is not good

```
Am I missing something? The text in these two blocks are exactly the same.

**Comment by parlay96** [comment]
If getserversideprops is used on page a, click the button router Push to page B，

There is a scroll bar on page a, for example, page a scrolls to the position of top 300

Then page B uses the browser return key to go to page a. Page B will appear. First scroll to top 300, that is, it will scroll to the position of the previous page a, and then render the dom of A

This is likely to be the problem of getserversideprops. There will be no problem if it is not used

**Reply by GabenGar** [reply]
> If getserversideprops is used on page a, click the button router Push to page B，

Are you using client-side routing options provided by NextJS?

**Comment by parlay96** [comment]
B returns to page a. If the a page getserversideprops is used, the return a page will appear. The browser has changed to the a path, but the DOM in the page is still the B page. Then Caton.

**Comment by parlay96** [comment]
I think it will have the same effect if you reproduce it yourself. Tell me how to solve the request

**Comment by parlay96** [comment]
I will thank you very much

**Comment by parlay96** [comment]
I think it will have the same effect if you reproduce it yourself. Tell me how to solve the request

**Comment by parlay96** [comment]
Are you using client-side routing options provided by NextJS?&nbsp; &nbsp;what do you mean




------------------&nbsp;原始邮件&nbsp;------------------
发件人: ***@***.***&gt;; 
发送时间: 2022年2月14日(星期一) 下午5:26
收件人: ***@***.***&gt;; 
抄送: ***@***.***&gt;; ***@***.***&gt;; 
主题: Re: [vercel/next.js] getServerSideProps BUG   Very serious (Discussion #34262)





  
If getserversideprops is used on page a, click the button router Push to page B，
  
Are you using client-side routing options provided by NextJS?
 
—
Reply to this email directly, view it on GitHub, or unsubscribe.
Triage notifications on the go with GitHub Mobile for iOS or Android.

**Comment by parlay96** [comment]
top

**Comment by parlay96** [comment]
I hope the next author can answer this question. Thank you

**Comment by parlay96** [comment]
`import type { NextPage } from 'next'
import Head from 'next/head'
import { useRouter } from "next/router"
import styles from '../styles/Home.module.css'

const Home: NextPage = () => {
  const router = useRouter()
  return (
    <div className={styles.container}>
      <Head>
        <title>Create Next App</title>
        <meta name="description" content="Generated by create next app" />
        <link rel="icon" href="/favicon.ico" />
      </Head>
        <div className={styles.content}>Take up space and make the page have scroll bars</div>
        <p style={{color:"red"}}>Go to the previous page and use the browser return key to return, which is different from the ordinary button. Try it</p>
        <button onClick={() =>router.push('/record')}>Click me to go to the record page</button>
        <br/>
        <div className={styles.content}>Take up space and make the page have scroll bars</div>
    </div>
  )
}

// Getserversideprops must exist on the return page
// Without using getserversideprops, there are no other problems
// The longer the getserversideprops request takes, the more it will get stuck when the previous page returns
export async function getServerSideProps(context) {
    // Leave a question, why push to the next page? When the browser comes back, it will be executed here
    // This will cause the page data to refresh when I return, which is not what I want
    console.log("router.push(/record), When you return to this page, it will trigger")
    return {
        props: {}, // will be passed to the page component as props
    }
}
export default Home
`

**Comment by parlay96** [comment]
`import type { NextPage } from 'next'
import Head from 'next/head'
import Image from 'next/image'
import styles from '../styles/Home.module.css'
import {useRouter} from "next/router";

const Record: NextPage = () => {
    const router = useRouter()
    return (
        <div className={styles.container}>
            Record page
            <div className={styles.content}>Take up space and make the page have scroll bars</div>
            <p style={{color:"red"}}>Clicking the return button of the browser has a different effect from clicking the button to return</p>
            <p style={{color:"red"}}>You will find that clicking on the upper left corner of the browser returns to the previous page. The previous page scroll bar returns to the top and the page flashes down</p>
            <p style={{color:"red"}}>Getserversideprops must exist on the return page</p>
            <p>Sometimes it will trigger. The result above is random</p>
            <button onClick={() => router.back()}>Click me back to the home page</button>
            <br />
            <div className={styles.content}>Take up space and make the page have scroll bars</div>
        </div>
    )
}

export default Record
`

**Comment by parlay96** [comment]
https://github.com/livelyPeng/next-demo   I wrote a replay demo
